### PR TITLE
chore: Add p1 haptics for selection actions

### PR DESCRIPTION
**Pull request by annacarey**
The type of this PR is: **chore**

This PR resolves [CX-550]

### Description

Added haptic props to `<Touchable>` and `Button` components in various parts of app according to [this doc](https://www.notion.so/artsy/Haptic-Opportunities-26eb8dfed6c94d9da80929496f8b32d1). (This PR covers only the priority P1 elements.) This follows the @pvinis' work in setup in the PR [here](https://github.com/artsy/eigen/pull/3835/files#diff-837e4859f1c1fabdbcc7aef271fc3b07cefe768572b18ed80aed10cd407dfa92R125).

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434
[CX-550]: https://artsyproduct.atlassian.net/browse/CX-550

### CODE REVIEW

**Review by pvinis — CHANGES_REQUESTED** [review]
Very nice!

I added one tiny suggestion, and as an extra thing, since we replace TouchableWithoutFeedback with our Touchable, I have a [commit here](https://github.com/artsy/eigen/pull/4137/commits/ec84bf6b1d8960ded7c846f996ce60dca48489cc) that you could cherry-pick, that would make the change in the tests not needed (keeping them with Touchable is fine too though), and most importantly it would keep the no feedback feel in on the buttons which would be nice.

File: src/lib/Scenes/Artwork/Components/FollowArtistButton.tsx [hunk 1]
```diff
@@ -58,11 +57,11 @@ export class FollowArtistButton extends React.Component<Props> {
   render() {
     const followButtonText = this.props.artist.is_followed ? "Following" : "Follow"
     return (
-      <TouchableWithoutFeedback onPress={() => this.handleFollowArtist()}>
+      <Touchable onPress={() => this.handleFollowArtist()} haptic="selection">
```
Comment (pvinis) [inline src/lib/Scenes/Artwork/Components/FollowArtistButton.tsx:60]: I think this and the other file that got TouchableWithoutFeedback replaced need `noFeedback`, and this PR is good to go!

File: src/lib/Components/AnimatedBottomButton.tsx [hunk 2]
```diff
@@ -39,9 +40,9 @@ export const AnimatedBottomButton: React.FC<AnimatedBottomButtonProps> = ({
 
   return (
     <BottomButtonContainer style={{ transform: [{ translateY: topOffset }] }}>
-      <TouchableHighlight activeOpacity={0.8} onPress={onPress} style={buttonStyles}>
+      <Touchable haptic="selection" activeOpacity={0.8} onPress={onPress} style={buttonStyles}>
```
[the remark this answers is not in the document]
Comment (annacarey) [reply src/lib/Components/AnimatedBottomButton.tsx]: @pvinis thanks for taking a look! I made this small change and pulled in your changes to `Touchable.ts` so then kept the tests as is (to check for `TouchableWithoutFeedback`.) Is there anything I need to do regarding your other PR now that some of the changes are in here?

[the remark this answers is not in the document]
Comment (pvinis) [reply src/lib/Components/AnimatedBottomButton.tsx]: no no, nothing to do. I'll rebase it later.

only add the noFeedback prop in the two components in this PR here that changed to Touchable and good to go 👍

File: src/lib/Scenes/Artwork/Components/FollowArtistButton.tsx (hunk 1 above)
Comment (annacarey) [reply src/lib/Scenes/Artwork/Components/FollowArtistButton.tsx:60]: I thought this button and the one in `ArtistSeriesMeta` are now supposed to have the selection feedback according to [our doc](https://www.notion.so/artsy/Haptic-Opportunities-26eb8dfed6c94d9da80929496f8b32d1). But maybe I'm misunderstanding?

Comment (pvinis) [reply src/lib/Scenes/Artwork/Components/FollowArtistButton.tsx:60]: ah. the noFeedback from the commit you cherry picker is no *visual* feedback. Touchable by default has highlight, but here we don't want it.

Comment (annacarey) [reply src/lib/Scenes/Artwork/Components/FollowArtistButton.tsx]: Got it! Updated those components with `noFeedback` and this passed in CI so should be good to go! Thanks @pvinis!

**Review by pvinis — APPROVED** [review]
noice! can't wait to feel the haptics on the next nightly.

**Pull request merged**